advanced/Scripts/piholeCheckout.sh

```diff
@@ -170,10 +170,23 @@ checkout() {
             echo "  ${TICK} Branch ${2} exists"
             echo "${2}" > /etc/pihole/ftlbranch
             chmod 644 /etc/pihole/ftlbranch
-            echo -e "  ${INFO} Switching to branch: \"${2}\" from \"${oldbranch}\""
+            echo -e "  ${INFO} Switching to branch \"${2}\" from \"${oldbranch}\""
             FTLinstall "${binary}"
             restart_service pihole-FTL
             enable_service pihole-FTL
+            str="Restarting FTL..."
+            echo -ne "  ${INFO} ${str}"
+            # Wait until name resolution is working again after restarting FTL,
+            # so that the updatechecker can run successfully and does not fail
+            # trying to resolve github.com
+            until getent hosts github.com &> /dev/null; do
```

Comment:
Is it possible `enable_service pihole-FTL` fails (or resolution fails) and we ended up in an infinite loop?

Reply:
I think it makes sense to loop until github.com is resolvable - if the network is down then there is no harm in waiting for it to be up again.